app/controllers/Tournament.scala

```diff
@@ -162,16 +165,20 @@ object Tournament extends LilaController {
       NoPlayban {
         val password = ctx.body.body.\("p").asOpt[String]
         negotiate(
-          html = repo enterableById id map {
-            case None => tournamentNotFound
+          html = repo enterableById id flatMap {
+            case None => fuccess(tournamentNotFound)
             case Some(tour) =>
-              env.api.join(tour.id, me, password)
-              Redirect(routes.Tournament.show(tour.id))
+              teamIdsIBelongTo(me) flatMap { teams =>
+                env.api.join(tour.id, me, password, teams)
```

Comment:
The idea of passing a function `User => Fu[List[TeamId]]` instead of `List[TeamId]` is that we only need to fetch the user team ids when the tournament requires it, which should be pretty rare